python/ray/serve/backend_worker.py

```diff
@@ -243,28 +245,6 @@ def _ray_serve_count_num_positional(self, f):
         # re-assign to default values
         serve_context.web = False
         serve_context.batch_size = None
-
-        # Tell router that current actor is idle
-        self._ray_serve_fetch()
+        self.mark_idle_in_router()
```

Comment:
we should remove this and change to router --push--> worker to reduce one rpc. the same logic can be applied when the response is successful returned from worker==worker is idle

Reply:
Agreed. I will do this in a follow-up